### ISSUE TITLE
Placeholder for the name of the decorated callable

### ISSUE THREAD

**Issue by crysanthus** [sighalt/logdecorator#11]
Is it possible to log the **decorated function name** just like in args or return value?
ie. from this to
```
@log_on_start(logging.DEBUG, "Starting process_text_files with {file_name:s}")
def process_text_files(file_name=None):
   pass
```
something like this,
```
@log_on_start(logging.DEBUG, "Starting {_function:s} with {file_name:s}")
def process_text_files(file_name=None):
   pass
```

**Comment by sighalt** [comment]
This is currently not supported. Could you explain the use case of such a feature a bit more? Because I don't see how the function name isn't a constant.

**Comment by crysanthus** [comment]
Completely a **nice to have** feature to make it easy to copy paste the decorator lines on new functions.
Thank you.

**Comment by sighalt** [comment]
I will have a look into this.

**Comment by mazz3rr** [comment]
Is there a way to use optional parameters in messages and then not pass them? From my experience this results in a KeyError. 
For instance, in the above example if `file_name` was not passed, that would result in error.

Sorry for off-topicking this thread, but I thought I could use the crysanthus's code as an example.

**Comment by sighalt** [comment]
@z33kz33k this is a bug, thanks for pointing it out. I created a new issue for that.

**Comment by mazz3rr** [comment]
Glad to help. This project rocks!

**Comment by sighalt** [comment]
Implemented and uploaded to PyPI. This feature is available from version 2.4.
